app/models/miq_request.rb

```diff
@@ -95,8 +94,11 @@ def resource_type
   end
 
   def initialize_attributes
-    self.requester_name ||= requester.name                         if requester.kind_of?(User)
-    self.requester ||= User.find_by_name(self.requester_name) if self.requester_name.kind_of?(String)
+    self.requester ||= User.find_by_userid(userid) if userid
+    self.requester ||= User.find_by_name(requester_name) if requester_name
+    self.requester_name ||= requester.try(:name)
+    self.userid ||= requester.try(:userid)
+    self.tenant ||= requester.try(:current_tenant)
```

Comment:
@kbrock As we discussed, can we completely get rid of setting `self.requester`?   This would also allow you to remove the `try` calls on the remaining lines here.